HLTrigger/Egamma/plugins/HLTEgammaCombMassFilter.cc

```diff
@@ -43,7 +45,35 @@ void HLTEgammaCombMassFilter::fillDescriptions(edm::ConfigurationDescriptions& d
 bool HLTEgammaCombMassFilter::hltFilter(edm::Event& iEvent,
                                         const edm::EventSetup& iSetup,
                                         trigger::TriggerFilterObjectWithRefs& filterproduct) const {
-  //right, issue 1, we dont know if this is a TriggerElectron, TriggerPhoton, TriggerCluster (should never be a TriggerCluster btw as that implies the 4-vectors are not stored in AOD)
+
+  using namespace trigger;
+  if (saveTags()) {
+    filterproduct.addCollectionTag(l1EGTag_);
+  }
+
+  edm::Handle<trigger::TriggerFilterObjectWithRefs> PrevFilterOutput;
+  iEvent.getByToken(firstLegLastFilterToken_, PrevFilterOutput);
+
+  std::vector<edm::Ref<reco::RecoEcalCandidateCollection> > phoCandsPrev;
+  PrevFilterOutput->getObjects(trigger::TriggerPhoton, phoCandsPrev);
+  std::vector<edm::Ref<reco::RecoEcalCandidateCollection> > clusCandsPrev;
+  PrevFilterOutput->getObjects(trigger::TriggerCluster, clusCandsPrev);
+  std::vector<edm::Ref<reco::ElectronCollection> > eleCandsPrev;
+  PrevFilterOutput->getObjects(trigger::TriggerElectron, eleCandsPrev);
+
+  if (!phoCandsPrev.empty()) {
+    for (auto& phoCand : phoCandsPrev) {
```

Comment:
Would `auto const&` work, instead ?

Reply:
Sure I will fix it, thanks for pointing this out

Reply:
This has been fixed accordingly in the latest commits.